src/GraphQL.SystemTextJson/GraphQLSerializer.cs

```diff
@@ -81,7 +81,7 @@ public GraphQLSerializer(Action<JsonSerializerOptions> configureSerializerOption
         /// <param name="serializerOptions">Specifies the JSON serializer settings</param>
         public GraphQLSerializer(JsonSerializerOptions serializerOptions)
         {
-            _options = serializerOptions ?? throw new ArgumentNullException(nameof(serializerOptions));
+            SerializerOptions = serializerOptions ?? throw new ArgumentNullException(nameof(serializerOptions));
 
             // TODO: fix this: it modifies serializerOptions
```

Comment:
@Shane32 Maybe it's time to fix TODO ?

For NSJ you wrote
![изображение](https://user-images.githubusercontent.com/21261007/149938510-68e0892a-bf6e-44fe-b202-840627782c53.png)

So maybe to write the same here and remove `ConfigureOptions(null);` ? + Add a few words in migration notes.

Reply:
\+ remove ` public GraphQLSerializer(JsonSerializerOptions serializerOptions, IErrorInfoProvider errorInfoProvider)`

Reply:
Well, we can't do that.  It is required by DI.  Besides, it has no bearing on this PR.

Reply:
**If** we don't want to modify passed serializer (or anything else passed from outside - DI) then we remove `ConfigureOptions` call and then `errorInfoProvider` argument become unused.

Reply:
I have no intention of changing the behavior in that respect.  I will be happy to remove the "todo" comments if you like.

Reply:
Serializer for NSJ is actualy the same API part as JsonSerializerOptions for STJ. So why we don't allow to modify first and allow to modify second?

Reply:
I would not modify the passed object (serializer or settings) for both cases.

Reply:
>  I will be happy to remove the "todo" comments if you like.

These TODO are necessary. I remember that the modification of these settings caused the error. Someone posted an issue.

Reply:
If you can explain how this will work with the `IOptions` DI pattern, I will consider your suggestions, but only in another PR.  I do not see any way this will work with the existing `IOptions` DI pattern.  If you want to remove support of the `IOptions` pattern, you can suggest it in another PR.

I agree that the constructor _should_ not alter incoming options classes.  I do not see a easy way to fix the problem.

Reply:
Keep in mind this "issue" applies equally to STJ and NJ.

Reply:
> If you want to remove support of the IOptions pattern

No. Why?

Reply:
#2865